src/WebJobs.Script/Host/ScriptHost.cs

```diff
@@ -428,9 +424,16 @@ protected virtual void Initialize()
 
                 LoadCustomExtensions();
 
+                // Do this after we've loaded the custom extensions. That gives an extension an opportunity to plug in their own implementations.
+                if (storageString != null)
+                {
+                    var lockManager = this.ScriptConfig.HostConfig.GetService<IDistributedLockManager>();
```

Comment:
Yes, this is committing further to allowing Script to pull the same services that SDK is using. 
SDK provides a default IDistributedLockManager.  In the on-prem case, an extension will override and inject their own. 

---
In reply to: [123672594](https://github.com/Azure/azure-webjobs-sdk-script/pull/1615#discussion_r123672594) [](ancestors = 123672594)